File: internal/content/testdata/bench/typescript/nc-permutations.ts
function permute(nums: number[]): number[][] {
    const res: number[][] = [];
    function backtrack(current: number[], remaining: number[]): void {
        if (remaining.length === 0) {
            res.push(current.slice());
            return;
        }
        for (let i = 0; i < remaining.length; i++) {
            current.push(remaining[i]);
            backtrack(current, remaining.slice(0, i).concat(remaining.slice(i + 1)));
            current.pop();
        }
    }
    backtrack([], nums);
    res.sort((a, b) => {
        for (let i = 0; i < a.length; i++) {
            if (a[i] !== b[i]) return a[i] - b[i];
        }
        return 0;
    });
    return res;
}

File: internal/content/testdata/bench/go/nc-permutations.go
import "sort"

func permute(nums []int) [][]int {
	res := [][]int{}
	var backtrack func(current, remaining []int)
	backtrack = func(current, remaining []int) {
		if len(remaining) == 0 {
			tmp := make([]int, len(current))
			copy(tmp, current)
			res = append(res, tmp)
			return
		}
		for i := range remaining {
			newRemaining := make([]int, 0, len(remaining)-1)
			newRemaining = append(newRemaining, remaining[:i]...)
			newRemaining = append(newRemaining, remaining[i+1:]...)
			backtrack(append(current, remaining[i]), newRemaining)
		}
	}
	backtrack([]int{}, nums)
	sort.Slice(res, func(i, j int) bool {
		a, b := res[i], res[j]
		for k := 0; k < len(a); k++ {
			if a[k] != b[k] {
				return a[k] < b[k]
			}
		}
		return false
	})
	return res
}
